pydicom/filewriter.py

```diff
@@ -414,6 +414,17 @@ def write_data_element(fp, data_element, encoding=default_encoding):
     if (hasattr(data_element, "is_undefined_length")
             and data_element.is_undefined_length):
         is_undefined_length = True
+        if data_element.tag == 0x7fe00010:  # pixel data
+            starts_with_item_tag = False
+            if fp.is_little_endian:
+                if data_element.value.startswith(b'\xfe\xff\x00\xe0'):
+                    starts_with_item_tag = True
+            else:
+                if data_element.value.startswith(b'\xff\xfe\xe0\x00'):
+                    starts_with_item_tag = True
+            if not starts_with_item_tag:
+                raise ValueError('Pixel Data with undefined length must '
+                                 'start with an an item tag')
     location = fp.tell()
```

Comment:
Can we add a comment at the top of this block to summarize why we are checking, and point to the issue?

Nitpicking here, but for fewer lines and easier to follow, perhaps set `val = data_element.value` first and use that to keep line length down, then just check the boolean directly like:
if fp.is_little_endian:
    starts_with_item_tag = val.startswith(...)
else:
   ...
then  also don't need to set `starts_with_item_tag` to False at the beginning.